Gems/Atom/Tools/ShaderManagementConsole/Scripts/GenerateShaderVariantListForAllShaders.py

```diff
@@ -0,0 +1,44 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+
+import re
+import json
+import GenerateShaderVariantListUtil
+
+def search_and_save_shadervariant(searchPath):
+    for subdir, dirs, files in os.walk(searchPath):
+        for file in files:
+            filename = os.path.join(subdir, file)
+            if re.search('^.*\.shader$', filename):
+                shaderVariantList, defaultShaderVariantListFilePath = GenerateShaderVariantListUtil.create_shadervariantlist_for_shader(filename)
+                if shaderVariantList.shaderVariants:
+                    azlmbr.shader.SaveShaderVariantListSourceData(defaultShaderVariantListFilePath, shaderVariantList)
```

Comment:
defaultShaderVariantListFilePath is (in most cases) not the right path. Instead of saving the .shadervariantlist next to the original shader in whatever gem it came from, we want to save the file in the active project's folder, because the list of variants for each project will be different. It's similar to how the SMC "Save As" dialog allows you to pick the original location or the project folder. The project folder should be the default.